cms/models.py

```diff
@@ -169,6 +170,15 @@ class ProgramIndexPage(CourseObjectIndexPage):
 
     def get_child_by_readable_id(self, readable_id):
         """Fetch a child page by the related Program's readable_id value"""
+        try:
+            queryset = ProgramRun.objects.annotate(
+                program_run=Concat("program__readable_id", models.Value("+"), "run_tag")
+            )
+            program_run = queryset.get(program_run=readable_id)
+            readable_id = program_run.program.readable_id
+        except ProgramRun.DoesNotExist:
+            pass
+
         return self.get_children().get(programpage__program__readable_id=readable_id)
```

Comment:
How does this program run identifier stored on the page for a checkout or other usage?

Reply:
that's not given in the ticket, just want to render the program like we were doing it before. I don't know if you have gone through the ticket or not.

```We should change our routing logic serve the normal program detail page if a program run id is provided in the URL```